shared/bsp/bsp_usb.h

```diff
@@ -25,26 +25,118 @@
 #include "usbd_cdc.h"
 #include "usbd_cdc_if.h"
 
-typedef void (*usb_callback_t)(uint8_t *buf, uint32_t len);
-
-/**
- * @brief register an arbitrary function to handle usb rx callback
- *
- * @param callback  a function pointer of type usb_callback_t
- */
-void usb_register_callback(const usb_callback_t callback);
-
-/**
- * @brief un-register usb rx callback function
- */
-void usb_unregister_callback(void);
-
-/**
- * @brief transmit data via usb
- *
- * @param buf buffer containing data
- * @param len length [in bytes] to transmit
- *
- * @return number of bytes successfully transmitted, -1 if usb busy, -2 if failed
- */
-int32_t usb_transmit(uint8_t *buf, uint32_t len);
+namespace bsp {
+
+class USB {
+  public:
+    /**
+     * @brief constructor for usb instance
+     *
+     * @param true if use callback for receiving, otherwise manual read is required. The
+     *        default value is false
+     *
+     * @note should not be called, to get the USB device, use bsp::GetUSBDevice() instead
```

Comment:
where is `GetUSBDevice`?

Reply:
This comment is written before I change the logic for storing the instance of USB device, and I forgot to delete the note after.